packages/near-membrane-dom/src/browser-realm.ts

```diff
@@ -76,6 +76,71 @@ function createDetachableIframe(): HTMLIFrameElement {
     return iframe;
 }
 
+// Create a default set of global own keys to be used for creating the shape of
+// any newly created realm's global object surface.
+const iframe = createDetachableIframe();
+const iframeWindow = ReflectApply(HTMLIFrameElementProtoContentWindowGetter, iframe, [])!.window;
+const defaultGlobalOwnKeys: OwnKeys = filterWindowKeys(getFilteredGlobalOwnKeys(iframeWindow));
+
+interface BrowserEnvironmentOptions {
+    distortionCallback?: DistortionCallback;
+    endowments?: PropertyDescriptorMap;
+    keepAlive?: boolean;
+    instrumentation?: Instrumentation | undefined;
+}
+
+interface RealmController {
+    cleanup: () => {};
+    evaluator: typeof eval;
+    needsExplicitCleanUp: boolean;
+}
+// @ts-ignore: TypeScript doesn't know what a ShadowRealm is yet.
+const { ShadowRealm: ShadowRealmCtor } = globalThis;
+let ShadowRealmProtoEvaluate: typeof ShadowRealmCtor.prototype.evaluate | undefined;
+
+if (typeof ShadowRealmCtor === 'function') {
+    ShadowRealmProtoEvaluate = ShadowRealmCtor.prototype.evaluate;
+}
+
+function createRealmController(options: BrowserEnvironmentOptions): RealmController {
+    let cleanup = () => {};
+    let evaluator: typeof eval;
+    let needsExplicitCleanUp = false;
+
+    if (typeof ShadowRealmCtor === 'function') {
+        const shadowRealm = new ShadowRealmCtor();
+        evaluator = ReflectApply(FunctionProtoBind, ShadowRealmProtoEvaluate, [shadowRealm]);
```

Comment:
I think this should be more strict considering that `evaluate` might have more arguments in the future.

Reply:
We don't restrict `window.eval` which in some engines takes a second param. This is used internally for our own wirings so restrictions seem extraneous.